src/main/java/org/opensearch/knn/jni/JNIService.java

```diff
@@ -23,6 +23,55 @@
  * Service to distribute requests to the proper engine jni service
  */
 public class JNIService {
+    private static final String FAISS_BINARY_INDEX_PREFIX = "B";
+
+    public static long initIndexFromScratch(long size, int dim, Map<String, Object> parameters, KNNEngine knnEngine) {
+        if (KNNEngine.FAISS == knnEngine) {
+            if (IndexUtil.isBinaryIndex(knnEngine, parameters)) {
+                return FaissService.initBinaryIndex(size, dim, parameters);
+            } else {
+                return FaissService.initIndex(size, dim, parameters);
+            }
+        }
+
+        throw new IllegalArgumentException(
+            String.format("initIndexFromScratch not supported for provided engine : %s", knnEngine.getName())
```

Comment:
Wherever you are using String.format pass Locale.ROOT as the first argument to ensure that correct locale is set for the exception

Reply:
I need to look into this. This would need to be changed in the already existing functions as this was copied from createIndex.

Reply:
Sure.